_code-samples/key-derivation/js/index.js

```diff
@@ -3,16 +3,49 @@
 // Organize imports
 const assert = require("assert")
 const brorand = require("brorand")
-const BN = require("bn.js")
+const { BigNumber } = require("bignumber.js")
 const elliptic = require("elliptic");
 const Ed25519 = elliptic.eddsa('ed25519');
 const Secp256k1 = elliptic.ec('secp256k1');
 const hashjs = require("hash.js");
-const Sha512 = require("ripple-keypairs/dist/Sha512")
+const isomorphic = require("@xrplf/isomorphic/sha512")
+const noble = require('@noble/curves/abstract/utils')
 const { codec, encodeAccountPublic, encodeNodePublic } = require("ripple-address-codec");
 
 const XRPL_SEED_PREFIX = 0x21
 
+class Sha512 {
```

Comment:
Would it be useful to update the `Sha512` class definition inside the JS client library? `packages/ripple-keypairs/src/Sha512.ts`

Would other projects/developers need similar changes?

Reply:
I don't think other developers would need those changes - I think that class is mostly an implementation detail of the library which supports the functions we do want to expose :)

Reply:
(This tutorial mostly mirrors the implementation details of xrpl.js to show how it works / why)